### DISCUSSION TITLE
How to click on a row selection to retrieve object from a table (not checkbox please)?

### DISCUSSION THREAD

**Discussion by KoichaDev** [TanStack/table#3558] [Q&A]
![image](https://i.ibb.co/yXvKdm8/2021-11-23-23-27-35.png)

I have seen a lot of examples where you can use a checkbox to click on a specific row to retrieve the object from it, but I am wondering if it's possible to click on an entire row instead to retrieve an object instead?

**Comment by ivan-sakoman** [comment]
you can work with a checkbox to retrieve an object but hide it with CSS?

**Comment by KoichaDev** [comment]
@ivan-sakoman I actually found a solution to do it:

The key is using the props that are called `getRowProps` as a callback for retrieving the data based on the onClick Handler. You can see it on code line number 126 when using the table component. The line number 66 is where the callback is added:
```js
 <tr {...row.getRowProps(getRowProps(row))}>
   {row.cells.map((cell) => {
     return <td {...cell.getCellProps()}>{cell.render("Cell")}</td>;
    })}
 </tr>
```

https://codesandbox.io/s/react-table-selector-on-click-for-the-row-getting-all-object-qhsfm
